src/ripple/core/JobCounter.h

```diff
@@ -159,8 +155,7 @@ class JobCounter
         // construct the wrapped job if we were not previously joined.
         // That would introduce a timing window.
         boost::optional<CountedJob<JobHandler>> ret {
-            CountedJob<JobHandler> {
-                *this, std::forward<JobHandler> (jobHandler)}};
+            CountedJob<JobHandler> {*this, std::move (handler)}};
```

Comment:
I believe we are unnecessarily constructing and destroying a CountedJob in that initialization. Consider the following alternative:
`        boost::optional<CountedJob<JobHandler>> ret;
        ret.emplace(*this, std::move(handler));`